release-notes/7.0/preview/api-diff/preview1/Microsoft.AspNetCore.App/7.0-preview1_Microsoft.Extensions.Configuration.md

```diff
@@ -0,0 +1,159 @@
+-        public static bool Exists(this IConfigurationSection section);
++        public static bool Exists(this IConfigurationSection? section);
+-        public static string GetConnectionString(this IConfiguration configuration, string name);
++        public static string? GetConnectionString(this IConfiguration configuration, string name);
+     }
+     public class ConfigurationKeyComparer : IComparer<string> {
+-        public int Compare(string x, string y);
++        public int Compare(string? x, string? y);
+     }
+     public sealed class ConfigurationManager : IConfiguration, IConfigurationBuilder, IConfigurationRoot, IDisposable {
+-        IDictionary<string, object> Microsoft.Extensions.Configuration.IConfigurationBuilder.Properties { get; }
++        IDictionary<string, object> IConfigurationBuilder.Properties { get; }
+-        IList<IConfigurationSource> Microsoft.Extensions.Configuration.IConfigurationBuilder.Sources { get; }
++        IList<IConfigurationSource> IConfigurationBuilder.Sources { get; }
+-        IEnumerable<IConfigurationProvider> Microsoft.Extensions.Configuration.IConfigurationRoot.Providers { get; }
++        IEnumerable<IConfigurationProvider> IConfigurationRoot.Providers { get; }
+-        public string this[string key] { get; set; }
++        public string? this[string key] { get; set; }
+-        IChangeToken Microsoft.Extensions.Configuration.IConfiguration.GetReloadToken();
++        IChangeToken IConfiguration.GetReloadToken();
+-        IConfigurationBuilder Microsoft.Extensions.Configuration.IConfigurationBuilder.Add(IConfigurationSource source);
++        IConfigurationBuilder IConfigurationBuilder.Add(IConfigurationSource source);
+-        IConfigurationRoot Microsoft.Extensions.Configuration.IConfigurationBuilder.Build();
++        IConfigurationRoot IConfigurationBuilder.Build();
+-        void Microsoft.Extensions.Configuration.IConfigurationRoot.Reload();
++        void IConfigurationRoot.Reload();
+     }
+     public static class ConfigurationPath {
+-        public static string GetParentPath(string path);
++        public static string? GetParentPath(string? path);
+-        public static string GetSectionKey(string path);
++        public static string? GetSectionKey(string? path);
+     }
+     public abstract class ConfigurationProvider : IConfigurationProvider {
+-        protected IDictionary<string, string> Data { get; set; }
++        protected IDictionary<string, string?> Data { get; set; }
+-        public virtual IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string parentPath);
++        public virtual IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string? parentPath);
+-        public virtual void Set(string key, string value);
++        public virtual void Set(string key, string? value);
+-        public virtual bool TryGet(string key, out string value);
++        public virtual bool TryGet(string key, out string? value);
```

Comment:
@eerhardt, same question here; presumably this should have `[NotNullWhen(true)]` on the `out string? value`?

Reply:
https://github.com/dotnet/runtime/pull/57401#discussion_r709918887